packages/sanity/src/core/store/_legacy/document/document-pair/utils/fetchFeatureToggle.ts

```diff
@@ -0,0 +1,35 @@
+import {type SanityClient} from '@sanity/client'
+import {map, type Observable, of, ReplaySubject, timer} from 'rxjs'
+import {catchError, share} from 'rxjs/operators'
+
+interface ActionsFeatureToggle {
+  actions: boolean
+}
+
+//in the "real" code, this would be observable.request, to a URI
+export const fetchFeatureToggle = (defaultClient: SanityClient): Observable<boolean> => {
+  const client = defaultClient.withConfig({apiVersion: 'X'})
+  const {dataset} = defaultClient.config()
+
+  return client.observable
+    .request({
+      uri: `/data/actions/${dataset}`,
+      withCredentials: true,
+    })
+    .pipe(
+      map((res: ActionsFeatureToggle) => res.actions),
+      catchError(() =>
+        // If we fail to fetch the feature toggle, we'll just assume it's disabled and fallback to legacy mutations
+        of(false),
+      ),
+      share({
+        // replay latest known state to new subscribers
+        connector: () => new ReplaySubject(1),
+        // this will typically be completed and unsubscribed from right after the answer is received, so we don't want to reset
+        resetOnRefCountZero: false,
+        // once the fetch has completed, we'll wait for 2 minutes before resetting the state.
+        // we'll then check again once a new subscriber comes in
+        resetOnComplete: () => timer(1000 * 120),
```

Comment:
This request seems to be sent every time I open a new document that hasn't already been opened in the past 120 seconds.

Is that the expected behaviour? I think the feature flag is controlled at the project level, rather than the document level, so it _seems_ wasteful (although ultimately inconsequential) to send the request for every document/listener.

Reply:
This behavior is a consequence of the fact that the various individual document pair methods are memoized on document ID and when they are initialized up, they start by checking whether to use the http actions APIs. One way to solve this would be to memoize the whole document store instead of the individual document methods, but that would be a bigger task, and probably not worth the effort this time around.

Reply:
Thank you for explaining! I agree it's probably not worthwhile to take it on at this time, but it would be nice 🙂.